### PR TITLE
Synchronizer L1 parallel:  improve context usage

### LINKED ISSUE #2567: synchronizer l1 sync: fix context usage
- Improve context usage inside  to be able to write unittest

### PR DESCRIPTION
**Pull request by joanestebanr**
Closes #2567
### What does this PR do?
- Improve context usage
- Remove start with starting block that was confusing
- Add unittest

Also modify config file (`test/config/test.node.config.toml`) to use new parallel proces mode:
```
UseParallelModeForL1Synchronization = true
	[Synchronizer.L1ParallelSynchronization]
		NumberOfParallelOfEthereumClients = 5
		CapacityOfBufferingRollupInfoFromL1 = 10
		TimeForCheckLastBlockOnL1Time = "5s"
		TimeoutForRequestLastBlockOnL1 = "60s"
		MaxNumberOfRetriesForRequestLastBlockOnL1 = 3
		TimeForShowUpStatisticsLog = "1m"
		TimeOutMainLoop = "5m"
		[Synchronizer.L1ParallelSynchronization.PerformanceCheck]
			AcceptableTimeWaitingForNewRollupInfo = "5s"
			NumIterationsBeforeStartCheckingTimeWaitinfForNewRollupInfo = 10
```

### Reviewers

Main reviewers:

- @ARR552

### CODE REVIEW

File: synchronizer/l1_rollup_info_producer.go [hunk 1]
```diff
@@ -195,11 +217,17 @@ func (l *l1RollupInfoProducer) verify() error {
...
+	log.Debug("producer: initialize")
 	err := l.verify()
 	if err != nil {
 		return err
 	}
+	if l.ctxParent != ctx || l.ctx != nil || l.cancelCtx == nil || (l.cancelCtx != nil && l.ctx.Err() != nil) {
+		log.Debug("producer: start called and need to create a new context")
+		l.ctxParent = ctx
+		l.ctx, l.cancelCtx = context.WithCancel(l.ctxParent)
+	}
```
Comment (ARR552) [inline synchronizer/l1_rollup_info_producer.go]:
```suggestion
	l.ctxParent = ctx
	if l.ctx != nil || l.cancelCtx == nil || (l.cancelCtx != nil && l.ctx.Err() != nil) {
		log.Debug("producer: start called and need to create a new context")
		l.ctx, l.cancelCtx = context.WithCancel(l.ctxParent)
	}
```

File: synchronizer/l1_worker_etherman.go [hunk 2]
```diff
@@ -108,32 +108,41 @@ type retrieveL1LastBlockResult struct {
...
 }
 
-func newWorker(etherman EthermanInterface) *worker {
-	return &worker{etherman: etherman, status: ethermanIdle}
+func newWorker(etherman EthermanInterface) *workerEtherman {
+	return &workerEtherman{etherman: etherman, status: ethermanIdle}
 }
 
-func (w *worker) asyncRequestRollupInfoByBlockRange(ctx context.Context, ch chan responseRollupInfoByBlockRange, wg *sync.WaitGroup, blockRange blockRange) error {
+func (w *workerEtherman) asyncRequestRollupInfoByBlockRange(ctx context.Context, cancelCtx context.CancelFunc, ch chan responseRollupInfoByBlockRange, wg *sync.WaitGroup, blockRange blockRange) error {
```
Comment (ARR552) [inline synchronizer/l1_worker_etherman.go]: Maybe we can pass a struct:
```
type ctxWithCancel struct {
   ctx       context.Context
   cancelCtx context.CancelFunc
}
```

**Review by ARR552 — APPROVED** [review]

File: synchronizer/l1_rollup_info_producer.go (hunk 1 above)
Comment (joanestebanr) [reply synchronizer/l1_rollup_info_producer.go]: I think that is not same: if `ctxParent` and incomming `ctx` we don't need to recreate contexts

File: synchronizer/l1_worker_etherman.go (hunk 2 above)
Comment (joanestebanr) [reply synchronizer/l1_worker_etherman.go]: I like very much the idea, I'll do the changes

**Pull request merged**